### DISCUSSION TITLE
3rd party script blocked by CORS

### DISCUSSION THREAD

**Discussion by AloisSeckar** [nuxt/scripts#193] [Q&A]
I am trying to call external script of Termly cookie consent service via Nuxt Scripts.

This is what I have in my `nuxt.config.ts`:

```
  scripts: {
    globals: {
      myScript: ['https://app.termly.io/resource-blocker/bbc9f62b-96e1-4064-bea5-e604bb669e55?autoBlock=on', {
        trigger: 'onNuxtReady',
      }],
    },
  },
```

However, the call gets blocked due to CORS:
![image](https://github.com/user-attachments/assets/d60dd8ab-472b-49e8-bb16-643d1db3e41d)

When I have script referenced from `useHead`, it worked with no issues:

```
useHead({
  script: [{ src: 'https://app.termly.io/resource-blocker/bbc9f62b-96e1-4064-bea5-e604bb669e55?autoBlock=on' }],
})
```

I am still getting confused with CORS stuff in general. Is there some setting, I need to set up for Nuxt Script module or for Nuxt in general to get this working? Or is it because the target domain refuses my requests due to something I cannot quite affect?

**Comment by harlan-zw** [comment]
Hi @AloisSeckar 

This is due to `useScript` applying strict privacy defaults, while it works for many third parties, some will block this privacy setting. 

You can get around this issue by doing 

```ts
 scripts: {
    globals: {
      myScript: ['https://app.termly.io/resource-blocker/bbc9f62b-96e1-4064-bea5-e604bb669e55?autoBlock=on', {
        trigger: 'onNuxtReady',
        crossorigin: false,
      }],
    },
  },
```

You can read more about this behaviour here: https://unhead.unjs.io/usage/composables/use-script#referrerpolicy-and-crossorigin

Let me know if you have any questions or feedback around this 🙂

**Reply by AloisSeckar** [reply]
Thank you for a quick help, but it doesn't seem to help me. The only difference, that I am getting HTTP Status 200, but still CORS error 🤨 
![image](https://github.com/user-attachments/assets/c1014db8-a61f-4990-adf2-056256418eb5)


Also my IDE says: 
```
Object literal may only specify known properties, and 'crossorigin' does not exist in type 'NuxtUseScriptOptionsSerializable'.
```

Isn't the syntax for Nuxt Scripts a little bit different perhaps? Or do I have some wrong version (Nuxt Scripts `0.6.5` and my Nuxt is `3.12.4`). 

Or it is because how Termly handle requests on their side now?

**Reply by AloisSeckar** [reply]
What I am still confused about: Does the error talk about header in request to 3rd party site I am sending (**I can add it somehow?**) or about header comming back from the service (**I can modify my app to ignore it?**)

**Comment by MuhammadM1998** [comment]
For anyone stumbling into this like I did, the correct way to do it is to use an object with `src` and other options as the first argument instead of the script's src string directly

```ts
const scripts = {
  globals: {
    myScript: [
      {
        src: 'https://app.termly.io/resource-blocker/bbc9f62b-96e1-4064-bea5-e604bb669e55?autoBlock=on',
        crossorigin: false,
        // ... Regular Script Tag Options
      },
      {
        trigger: 'onNuxtReady',
        // ... Other Nuxt Scripts Options
      },
    ],
  },
}
```

**Comment by harlan-zw** [comment]
Going to close this — the `crossorigin: false` workaround above is the documented escape hatch when a third-party blocks CORS on the script tag. Open a new issue if anything else surfaces 🙏
